### PR TITLE
Add travis.yml and script to validate templates

### LINKED ISSUE #54: Suggestion: CI for the templates
## Expected Behaviour

The templates should validate via a travis build. Whenever someone raises a PR, all templates should be used to create the default sample, build it and run it checking the result

## Current Behaviour

Manually verified.

## Possible Solution

A shell script should be able to do this. Functions can be run using docker run rather than using full OpenFaaS.

Automation would be ideal or some kind of config file. This should be minimal as to not bloat the repository.

### PR DESCRIPTION
**Pull request by burtonr**
Signed-off-by: Burton Rheutan <rheutan7@gmail.com>

## Description

This adds a script that will be run with a CI system to
validate the templates are able to build and output
a value verifying that they are functional in the most
basic functionality of returning a value

## Motivation and Context

- [X] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## Which issue(s) this PR fixes 

Fixes #54 

## How Has This Been Tested?

Ran script locally:
```
./verfiy.sh
```
Also, ran build via [TravisCI](https://travis-ci.org/burtonr/templates/builds/410620475)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by burtonr** [comment]
@alexellis, @LucasRoesler , and @ewilde Thank you for your early reviews and suggestions!

This is ready for a final review. Please let me know if there's anything I missed from any of your suggestions

File: verify.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        exit 1
+    else
+        echo $image template validation successful
+    fi
+}
+
+if ! [ -x "$(command -v faas-cli)" ]; then
+    HERE=`pwd`
+    cd /tmp/
+    curl -sL https://github.com/openfaas/faas-cli/releases/download/0.6.9/faas-cli > faas-cli
```
Comment (LucasRoesler) [inline verify.sh]: It would be nice if the version was a variable

File: verify.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+set -e
+
+CLI="faas-cli"
+func_name="validate"
```
Comment (LucasRoesler) [inline verify.sh]: I wasnt what this was and it took me a few minutes to figure out, might be nice to have a comment explaining `func_name`

File: verify.sh (hunk 1 above)
Comment (alexellis) [reply verify.sh]: It should use the installation script for Linux. Could also be a pre-install step for Travis

File: verify.sh [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+cd ./template
+
+# verify each of the templates
+for dir in ./*/
+do
+    dirname=${dir%*/}
+    template=${dirname##*/}
+    # skip dockerfile template
+    if [ "$template" = "dockerfile" ] ; then continue ; fi ;
```
Comment (alexellis) [inline verify.sh]: We can test this one too, it also has customisations including non-root etc.

File: verify.sh (hunk 2 above)
Comment (alexellis) [inline verify.sh]: My suggestion would be to build with the name of the function being unique and based upon the folder name - i.e. `python3-ci` / `go-ci`, only delete from the filesystem at the very end, if there was no error detected.

**Comment by burtonr** [comment]
@alexellis  and @LucasRoesler Thanks for your input! I've pushed an update that addresses your comments.

**Comment by alexellis** [comment]
Let's try it out..

**Comment by alexellis** [comment]
https://travis-ci.org/openfaas/templates/builds/411347283

**Pull request merged**